public/pages/QueryInsights/QueryInsights.tsx

```diff
@@ -0,0 +1,163 @@
+    },
+    {
+      field: 'indices',
+      name: 'Indices',
+      render: (indices: string[]) => indices.toString(),
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'search_type',
+      name: 'Search type',
+      render: (searchType: string) => searchType.replaceAll('_', ' '),
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'node_id',
+      name: 'Coordinator node ID',
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'total_shards',
+      name: 'Total shards',
+      sortable: true,
+      truncateText: true,
+    },
+  ];
+
+  const retrievedQueries: any[] = [];
+  const [queries, setQueries] = useState(retrievedQueries);
+
+  const defaultStart = 'now-24h';
+  const [recentlyUsedRanges, setRecentlyUsedRanges] = useState([
+    { start: defaultStart, end: 'now' },
+  ]);
+  const [currStart, setStart] = useState(defaultStart);
+  const [currEnd, setEnd] = useState('now');
+
+  const parseDateString = (dateString: string) => {
+    const date = dateMath.parse(dateString);
+    return date ? date.toDate().getTime() : new Date().getTime();
+  };
+
+  const updateQueries = ({ start, end }: { start: string; end: string }) => {
+    const startTimestamp = parseDateString(start);
+    const endTimestamp = parseDateString(end);
+    setQueries(
+      retrievedQueries.filter(
+        (item) => item.timestamp >= startTimestamp && item.timestamp <= endTimestamp
+      )
+    );
+  };
+
+  const onTimeChange = ({ start, end }: { start: string; end: string }) => {
+    const usedRange = recentlyUsedRanges.filter(
+      (range) => !(range.start === start && range.end === end)
+    );
+    usedRange.unshift({ start, end });
+    setStart(start);
+    setEnd(end);
+    setRecentlyUsedRanges(usedRange.length > 10 ? usedRange.slice(0, 9) : usedRange);
+    updateQueries({ start, end });
+  };
+
+  const onRefresh = async ({ start, end }: { start: string; end: string }) => {
+    updateQueries({ start, end });
+  };
+
+  useEffect(
+    () => {
+      onRefresh({ start: currStart, end: currEnd });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  return (
+    <div>
+      <EuiInMemoryTable
+        items={queries}
+        columns={cols}
+        sorting={{
+          sort: {
+            field: 'timestamp',
+            direction: 'desc',
+          },
+        }}
+        search={{
+          box: {
+            placeholder: 'Search queries',
+            schema: false,
+          },
+          toolsRight: [
+            <EuiSuperDatePicker
+              start={currStart}
+              end={currEnd}
+              recentlyUsedRanges={recentlyUsedRanges}
+              onTimeChange={onTimeChange}
+              onRefresh={onRefresh}
+              updateButtonProps={{ fill: false }}
+            />,
+          ],
+        }}
+        executeQueryOptions={{
+          defaultFields: [
+            'timestamp',
```

Comment:
These strings are used in 
```
 const cols: Array<EuiBasicTableColumn<any>> = [
    {
      field: 'timestamp',
      name: 'Time stamp',
...
```
as well. We should consider create Constants variables for them.